src/math/CMakeLists.txt

```diff
@@ -5,31 +5,41 @@ if(BUILD_LIBRARY)
 	return()
 endif()
 
+is_zephyr(it_is_zephyr)
+
 add_local_sources(sof numbers.c)
 
-if(CONFIG_CORDIC_FIXED)
+if(CONFIG_CORDIC_FIXED OR it_is_zephyr)
```

Comment:
`is_zephyr` is only a check for CONFIG_ZEPHYR_SOF_MODULE. The indirection is for brevity, flexibility and "future-proofness".

Reply:
why not just use ```CONFIG_ZEPHYR_SOF_MODULE``` if they are equivalent.

Reply:
They are equivalent now but who knows in the future. I'm currently spreading this conditional across almost all `sof/**/CMakeLists.txt` files, so this very thin indirection protects us against any future change. If this CONFIG_ namve ever changes in Zephyr in the future we could even make the macro support both the old and new for a seamless transition.

Reply:
If they are equivalent now then pls use, otherwise this is really just unneeded indirection today.

Reply:
It's a bit late now, this `is_zephyr()` macro was reviewed, approved and added in #8422 and is already in use in `src/init/`, `src/ipc` and `lmdk/cmake`. So not using it would be inconsistent. 

Speaking of consistency, another aspect of the problem is that there are at least two macros: `CONFIG_SOF` and `CONFIG_ZEPHYR_SOF_MODULE`. Which one is "best"? Maybe there's even a another and better one, who knows. I looked in the documentation and in the source a bit but no one really knows for sure; it's not a common problem to solve. Hence the indirection which can be fixed with a one-line change instead of a mass search/replace. Again this was discussed in #8422 cc: @andyross

Reply:
Tend to agree that the existing kconfigs aren't super well named for this purpose.  They're artifacts of the way the upstream Zephyr module system works: ZEPHYR_SOF_MODULE is automatically generated, and 'config SOF' is added in the Kconfig for the module.  FWIW I NAK'd the latter because in this context it means the opposite of what it says ("CONFIG_SOF" might be expected to mean "xtos/traditional SOF and not Zephyr", when it means the opposite).

My personal vote would be a SOF-internal API that contains "zephyr" in the name like this one.  Whether it's a kconfig or a cmake macro doesn't seem all that important I guess?  Would be easy to fix in a different patch too.

Reply:
`CONFIG_CORDIC OR CONFIG_ZEPHYR_...` would also give the wrong impression that both are choices, see discussion below. The second one is obviously not a CONFIGuration choice.

Reply:
> For Zephyr stuff, we need something CONFIG_SOF_ZEPHYR_X the precedence being this is a SOF Kconfig and NOT a Zephyr Kconfig.
> @marc-hb can you incrementally change the is_zephyr to use the Kconfig, this way is easier for everyone to follow.

You lost me sorry.

Reply:
I just found that some places started using `CONFIG_ZEPHYR_NATIVE_DRIVERS`...